src/llscan.cc

```diff
@@ -0,0 +1,470 @@
+
+    inspect_options.detailed = detailed_;
+
+    char** start = ParseInspectOptions(cmd, &inspect_options);
+
+    std::string full_cmd;
+    for (; start != nullptr && *start != nullptr; start++) full_cmd += *start;
+
+    SBExpressionOptions options;
+    SBValue value = target.EvaluateExpression(full_cmd.c_str(), options);
+    if (value.GetError().Fail()) {
+        SBStream desc;
+        if (!value.GetError().GetDescription(desc)) return false;
+        result.SetError(desc.GetData());
+        return false;
+    }
+
+    // Load V8 constants from postmortem data
+    llv8.Load(target);
+
+    v8::Value v8_mapaddr(&llv8, value.GetValueAsSigned());
+    v8::Error err;
+
+    if (err.Fail()) {
+        result.SetError("Failed to evaluate expression");
+        return false;
+    }
+
+    uint64_t mapaddr = value.GetValueAsUnsigned();
+    std::map<uint64_t, TypeRecord*>::iterator instance_it =
+            mapstoinstances->find(mapaddr);
+    if (instance_it != mapstoinstances->end()) {
+        TypeRecord* t = instance_it->second;
+        for (std::set<uint64_t>::iterator it = t->instances->begin();
+                it != t->instances->end(); ++it) {
+            v8::Error err;
+            v8::Value v8_value(&llv8, *it);
+            std::string res = v8_value.Inspect(&inspect_options, err);
+            result.Printf("%s\n", res.c_str());
+        }
+
+    } else {
+        result.Printf("No objects found with map %" PRIx64 "\n", mapaddr);
+    }
+
+    return true;
+
+}
+
+FindJSObjectsVisitor::FindJSObjectsVisitor(SBTarget* target,
+        SBCommandReturnObject* result) {
+    _target = target;
+    _result = result;
+    // Load V8 constants from postmortem data
+    llv8.Load(*target);
+    _address_byte_size = _target->GetProcess().GetAddressByteSize();
+    _found_count = 0;
+    mapstoinstances = new std::map<uint64_t, TypeRecord*>();
+}
+
+/* Visit every address, a bit brute force but it works. */
+uint64_t FindJSObjectsVisitor::Visit(uint64_t location, uint64_t available) {
+
+    lldb::SBError error;
+
+    // Test if the map points to a real map.
+    // Try to create an object out of it.
+
+    uint64_t word = _target->GetProcess().ReadUnsignedFromMemory(location,
+            _target->GetProcess().GetAddressByteSize(), error);
+
+    v8::Value v8_value(&llv8, word);
+
+    v8::Error err;
+    // Test if this is SMI
+    // Skip inspecting things that look like Smi's, they aren't objects.
+    llnode::v8::Smi smi(v8_value);
+    if (smi.Check()) {
+        return _address_byte_size;
+    }
+
+    llnode::v8::HeapObject heap_object(v8_value);
+    if (!heap_object.Check()) {
+        return _address_byte_size;
+    }
+    if (heap_object.IsHoleOrUndefined(err)) {
+        return _address_byte_size;
+    }
+    if (err.Fail()) {
+        return _address_byte_size;
+    }
+
+    llnode::v8::HeapObject map_object = heap_object.GetMap(err);
+    if (err.Fail() || !map_object.Check()) {
```

Comment:
I think `GetMap` calls `Check` automatically through the `LoadFieldValue<HeapObject>`.